packages/wallet/src/KeyManagement/util/mapHardwareSigningData.ts

```diff
@@ -0,0 +1,547 @@
+            const multiHostName = relay.as_multi_host_name();
+            const dnsName = multiHostName?.dns_name().record();
+            if (dnsName) {
+              ledgerRelays.push({
+                params: { dnsName },
+                type
+              });
+            }
+          }
+        }
+      }
+      const cost = params?.cost().to_str();
+      const margin = params?.margin();
+      const pledge = params?.pledge().to_str();
+
+      const operator = Buffer.from(params.operator().to_bytes()).toString('hex');
+      let poolKey: PoolKey;
+      if (operator === rewardAccountKeyHash) {
+        signingMode = TransactionSigningMode.POOL_REGISTRATION_AS_OPERATOR;
+        poolKey = {
+          params: { path: rewardAccountKeyPath },
+          type: PoolKeyType.DEVICE_OWNED
+        };
+      } else {
+        poolKey = {
+          params: { keyHashHex: operator },
+          type: PoolKeyType.THIRD_PARTY
+        };
+      }
+
+      const poolMetadata = params.pool_metadata();
+      const metadata = poolMetadata
+        ? {
+            metadataHashHex: Buffer.from(poolMetadata.pool_metadata_hash().to_bytes()).toString('hex'),
+            metadataUrl: poolMetadata.url().url()
+          }
+        : null;
+      const rewardAccountHex = Buffer.from(params.reward_account().to_address().to_bytes()).toString('hex');
+      const rewardAccount: PoolRewardAccount =
+        rewardAccountHex === addressHex
+          ? {
+              params: { path: rewardAccountKeyPath },
+              type: PoolRewardAccountType.DEVICE_OWNED
+            }
+          : {
+              params: { rewardAccountHex },
+              type: PoolRewardAccountType.THIRD_PARTY
+            };
+      const vrfKeyHashHex = Buffer.from(params.vrf_keyhash().to_bytes()).toString('hex');
+
+      certificate.params = {
+        cost,
+        margin: {
+          denominator: margin.denominator().to_str(),
+          numerator: margin.numerator().to_str()
+        },
+        metadata,
+        pledge,
+        poolKey,
+        poolOwners,
+        relays: ledgerRelays,
+        rewardAccount,
+        vrfKeyHashHex
+      };
+    }
+    certs.push(certificate);
+  }
+  return {
+    certs,
+    signingMode
+  };
+};
+
+const prepareLedgerWithdrawals = (withdrawals: CSL.Withdrawals, rewardAccountKeyPath: BIP32Path): Withdrawal[] => {
+  const ledgerWithdrawals = [];
+  for (let i = 0; i < withdrawals.keys().len(); i++) {
+    const withdrawal = { stakeCredential: {} } as Withdrawal;
+    const rewardAddress = withdrawals.keys().get(i);
+    const paymentCredentials = rewardAddress.payment_cred();
+    const paymentCredentialsScriptHash = paymentCredentials.to_scripthash();
+    if (rewardAddress.payment_cred().kind() === 0) {
+      withdrawal.stakeCredential.type = StakeCredentialParamsType.KEY_PATH;
+      // Ignore! KeyStakeCredentialParams / ScriptStakeCredentialParams are not exported
+      // @ts-ignore
```

Comment:
What are the errors for these @ts-ignores and are we sure the code is correct? Maybe just coerce type with `as any` instead?